testsuite/tests/ui/oas/test_active_docs_v3.py

```diff
@@ -0,0 +1,126 @@
+"""Tests for active docs OAS3"""
+import json
+
+import importlib_resources as resources
+import pytest
+from threescale_api.resources import Service
+
+from testsuite import rawobj
+from testsuite.ui.views.admin.audience.developer_portal import ActiveDocsNewView
+from testsuite.ui.views.admin.product.active_docs import ActiveDocsDetailView
+from testsuite.utils import blame
+
+
+@pytest.fixture(scope="module")
+def service_settings2(request):
+    """settings for second service"""
+    return {"name": blame(request, "svc"),
+            "backend_version": Service.AUTH_APP_ID_KEY}
+
+
+@pytest.fixture(scope="module")
+def service2(backends_mapping, custom_service, service_settings2, service_proxy_settings, lifecycle_hooks):
+    """Second service to test with"""
+    return custom_service(service_settings2, service_proxy_settings, backends_mapping, hooks=lifecycle_hooks)
+
+
+@pytest.fixture(scope="module")
+def active_doc2(request, service2, oas3_body, custom_active_doc):
+    """Active doc. bound to second service."""
+    return custom_active_doc(rawobj.ActiveDoc(blame(request, "activedoc"), oas3_body, service=service2),
+                             autoclean=False)
+
+
+@pytest.fixture(scope="module")
+def oas3_spec():
+    """
+    OAS3 Active doc spec with 3scale `user key` parameter
+    """
+    user_key = {
+        "name": "user_key",
+        "description": "Your access API Key",
+        "type": "string",
+        "in": "query",
+        "x-data-threescale-name": "user_keys",
+        "required": True
+    }
+    oas_spec = resources.files("testsuite.resources.oas3").joinpath("petstore-expanded.json").read_text()
+    json_spec = json.loads(oas_spec)
+    json_spec['paths']['/pets']['get']['parameters'].append(user_key)
+    oas_spec = json.dumps(json_spec)
+    return oas_spec
+
+
+@pytest.fixture(scope="module")
+def prod_client(prod_client):
```

Comment:
If you use prod_client you need to mark this test as disruptive. Why do you need production gateway in the first place?

Reply:
Because oas3 active docs make request against production endpoint of service.

Reply:
Ok, then you can either mark this test as disruptive or run it with self-managed APIcast